js/graviex.js

```diff
@@ -0,0 +1,1073 @@
+        const result = [];
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            const id = this.safeString (market, 'id');
+            const symbol = this.safeString (market, 'name');
+            const [ baseId, quoteId ] = symbol.split ('/');
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': this.safeCurrencyCode (baseId),
+                'quote': this.safeCurrencyCode (quoteId),
+                'settle': undefined,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': undefined,
+                'type': 'spot',
+                'spot': true,
+                'margin': false,
+                'swap': false,
+                'future': false,
+                'option': false,
+                'active': undefined,
+                'contract': false,
+                'linear': undefined,
+                'inverse': undefined,
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': this.precision,
+                'limits': this.extend ({
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                }, this.limits),
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        // {
+        //     "name": "GIO/BTC",
+        //     "base_unit": "gio",
+        //     "base_fixed": 4,
+        //     "base_fee": 0.002,
+        //     "quote_unit": "btc",
+        //     "quote_fixed": 8,
+        //     "quote_fee": 0.002,
+        //     "api": true,
+        //     "base_lot": null,
+        //     "quote_lot": null,
+        //     "base_min": "0.00000010",
+        //     "quote_min": "0.00000010",
+        //     "blocks": 260722,
+        //     "block_time": "2022-01-31 14:10:08",
+        //     "wstatus": "on",
+        //     "low": "0.0000005",
+        //     "high": "0.00000058",
+        //     "last": "0.00000058",
+        //     "open": "0.000000510",
+        //     "volume": "70673.0412",
+        //     "volume2": "0.03837782973",
+        //     "sell": "0.00000058",
+        //     "buy": "0.00000055",
+        //     "at": 1643628427,
+        // },
+        const timestamp = this.milliseconds ();
+        const marketId = this.safeString (ticker, 'name');
+        const symbol = this.safeSymbol (marketId, market);
+        const last = this.safeFloat (ticker, 'last');
+        return this.safeTicker ({
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (ticker, 'high'),
```

Comment:
we no long use safeFloat... we use safeNumber

Reply:
however inside of safeTicker we should use `safeString`

Reply:
OKay. Fixed in the latest commit https://github.com/ccxt/ccxt/commit/24b2627beff01a76b81a32b5728efb6846cd8010